runtime-modules/project-token/src/types.rs

```diff
@@ -0,0 +1,337 @@
+use codec::{Decode, Encode};
+use frame_support::{
+    dispatch::{fmt::Debug, DispatchResult},
+    ensure,
+};
+use sp_arithmetic::traits::{Saturating, Zero};
+use sp_runtime::traits::{Convert, Hash};
+use sp_std::collections::btree_map::{BTreeMap, Iter};
+use sp_std::iter::Sum;
+
+/// Info for the account
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub struct AccountData<Balance> {
+    /// Non-reserved part of the balance. There may still be restrictions
+    /// on this, but it is the total pool what may in principle be
+    /// transferred, reserved and used for tipping.
+    pub(crate) free_balance: Balance,
+
+    /// This balance is a 'reserve' balance that other subsystems use
+    /// in order to set aside tokens that are still 'owned' by the
+    /// account holder, but which are not usable in any case.
+    pub(crate) reserved_balance: Balance,
+}
+
+/// Info for the token
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Default, Debug)]
+pub struct TokenData<Balance, Hash, BlockNumber> {
+    /// Current token issuance
+    pub(crate) supply: Balance,
+
+    /// Initial issuance state
+    pub(crate) issuance_state: OfferingState,
+
+    /// Transfer policy
+    pub(crate) transfer_policy: TransferPolicy<Hash>,
+
+    /// Symbol used to identify token
+    pub(crate) symbol: Hash,
+
+    /// Patronage Information
+    pub(crate) patronage_info: PatronageData<Balance, BlockNumber>,
+}
+
+/// Patronage information, patronage configuration = set of values for its fields
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Default, Debug)]
+pub struct PatronageData<Balance, BlockNumber> {
+    /// Patronage rate
+    pub(crate) rate: Balance,
+
+    /// Tally count for the outstanding credit before latest patronage config change
+    pub(crate) unclaimed_patronage_tally_amount: Balance,
+
+    /// Last block the patronage configuration was updated
+    pub(crate) last_unclaimed_patronage_tally_block: BlockNumber,
```

Comment:
I think so far the best name is simply `last_tally_update_block` @bedeho. 
Since we want to update the `tally_amount` whenever:
- rate changes
- outstanding supply changes
- patronage is claimed

Reply:
I think this name is fine, it is non-ambiguous. It was suggested by @bedeho who is not against using long names